scala2-functional-program-design/week3/codecs/src/main/scala/codecs/codecs.scala

```diff
@@ -202,16 +206,21 @@ trait DecoderInstances {
     */
   implicit def listDecoder[A](implicit decoder: Decoder[A]): Decoder[List[A]] =
     Decoder.fromFunction {
-      ???
+      json => json match {
+        case Json.Arr(arr) => Option(arr.foldLeft[List[A]](List[A]())((z: List[A], j: Json) => z:::(if (decoder.decode(j).isEmpty) List[A]() else List[A](decoder.decode(j).get))))
```

Comment:
Ohh jeezzz... 😆 
Let's try implementing this again, simple.

Reply:
Fix, will be far nicer, I assumed fold is a must since we must return only successful decodes <- assumption is wrong, code changed accordingally

Reply:
Much better :)